server/src/test/java/org/elasticsearch/cluster/routing/allocation/decider/DiskThresholdDeciderTests.java

```diff
@@ -930,8 +915,8 @@ public void testCanRemainWithShardRelocatingAway() {
             decision.getExplanation(),
             containsString(
                 "the shard cannot remain on this node because it is above the high watermark cluster setting "
-                    + "[cluster.routing.allocation.disk.watermark.high=70%] and there is less than the required [30.0%] free disk on node, "
-                    + "actual free: [20.0%]"
+                    + "[cluster.routing.allocation.disk.watermark.high=70%] and there is less than the required [30b] free space "
+                    + "on node, actual free: [20b], actual used: [80%]"
```

Comment:
I like that these messages now show the absolute byte values, but I think we shouldn't lose the percentages if the user is defining their watermarks in those terms. If it's too hard to make the message depend on whether the watermark is absolute or not then I would be ok with including both byte values and percentages in all such messages.

Reply:
With the headroom settings, it will be sometimes difficult to go back in figuring out which setting prevailed (the percentage or the headroom). So, I edited the message to include all relevant information that can be used to understand what happened. 

### Case where percentage prevails
This is the case of this example. You see that `actual used: [80%]` is over the `[cluster.routing.allocation.disk.watermark.high=70%]` watermark. So this should be fine. I do not think we also need the free percentage, since it is not used for the watermark calculation (the used percentage is over the watermark).

### Case where max headroom prevails
It will read something like:
```
[cluster.routing.allocation.disk.watermark.frozen.max_headroom=20gb] and there is less than the required [20gb] free space on node, actual free: [20b], actual used: [99%]"
```
This should also be fine. It shows that the max_headroom is violated since there is less than 20gb free space. We do not need a free percentage here because it is not taken into account (the bytes and headroom is taken into account).

Please tell me if this is not sufficient and unresolve the conversation.

Reply:
Ok I see now, yes this makes sense to me.